### PR TITLE
Add an env variable to the release builder to tell mach to build with debug assertions.

### PR DESCRIPTION
**Pull request by emilio**
r? @aneeshusa

---
This change is [Reviewable](https://reviewable.io/reviews/servo/saltfs/480)

### CODE REVIEW

File: buildbot/master/files/config/master.cfg [hunk 1]
```diff
@@ -170,7 +170,7 @@ c['builders'] = [
     DynamicServoBuilder("arm64", CROSS_SLAVES, envs.build_arm64),
     DynamicServoBuilder("linux-dev", LINUX_SLAVES, envs.build_linux),
     DynamicServoBuilder("linux-nightly", LINUX_SLAVES, envs.build_linux),
-    DynamicServoBuilder("linux-rel", LINUX_SLAVES, envs.build_linux),
+    DynamicServoBuilder("linux-rel", LINUX_SLAVES, envs.build_linux_rel_debug_assert),
```
Comment (aneeshusa) [inline buildbot/master/files/config/master.cfg]: You'll need to move the last argument to the next line to fit inside the line length limit.

Comment (emilio) [inline buildbot/master/files/config/master.cfg]: Gah, good catch, thanks :)

Done

File: buildbot/master/files/config/master.cfg [hunk 2]
```diff
@@ -170,7 +170,8 @@ c['builders'] = [
     DynamicServoBuilder("arm64", CROSS_SLAVES, envs.build_arm64),
     DynamicServoBuilder("linux-dev", LINUX_SLAVES, envs.build_linux),
     DynamicServoBuilder("linux-nightly", LINUX_SLAVES, envs.build_linux),
-    DynamicServoBuilder("linux-rel", LINUX_SLAVES, envs.build_linux),
+    DynamicServoBuilder("linux-rel", LINUX_SLAVES
```
Comment (aneeshusa) [inline buildbot/master/files/config/master.cfg]: You'll need a comma after this.

Comment (emilio) [inline buildbot/master/files/config/master.cfg]: Pffft... Thanks, done, should learn to be more careful when pushing fixups, sigh.

Are there docs in how to install the required deps and test changes to this file, btw?

Comment (aneeshusa) [inline buildbot/master/files/config/master.cfg]: It's partially documented right now (the Vagrant-related sections [on the wiki](https://github.com/servo/servo/wiki/Buildbot-administration#making-changes)), but the test suite isn't well documented right now. I have a half written draft sitting on my hard drive somewhere...

**Comment by jdm** [comment]
I'm a bit nervous that whenever we next update the builders to use this we'll have permafailures that we've never seen before...

**Comment by emilio** [comment]
@jdm: The way we're doing this is effectively to avoid it. This env var currently does nothing.

Once this is rolled out, I need to add a `mach` PR using that env var to add the `--cfg` flag to the rust flags, and then cry with the failures, and fix it, or figure out what we can do about it :)

**Pull request merged**